indico/modules/rb_new/operations/rooms.py

```diff
@@ -198,3 +201,18 @@ def search_for_rooms(filters, availability=None):
                                             db.or_(booking_limit_days.is_(None),
                                                    selected_period_days <= booking_limit_days))))
     return query
+
+
+def get_room_events(room, start_dt, end_dt, repeat_frequency, repeat_interval):
+    occurrences = ReservationOccurrence.create_series(start_dt, end_dt, (repeat_frequency, repeat_interval))
+    excluded_categories = rb_settings.get('excluded_categories')
+    events = (Event.query
+              .filter(~Event.is_deleted,
+                      Event.own_room == room,
+                      db.or_(Event.happens_between(as_utc(occ.start_dt), as_utc(occ.end_dt)) for occ in occurrences),
+                      Event.timezone == config.DEFAULT_TIMEZONE,
+                      db.and_(Event.category_id != cat['id'] for cat in excluded_categories))
+              .options(subqueryload('acl_entries'))).all()
+    user_events = [event for event in events if [entry for entry in event.acl_entries
```

Comment:
Please don't iterate over `acl_entries` manually - we have shortcuts for this, i.e. `event.can_manage(session.user)`

If this is for performance reasons to skip group-based access, then do the filtering directly in the query (and remove the subqueryload:

```py
                      Event.acl_entries.any(db.and_(EventPrincipal.type == PrincipalType.user,
                                                    EventPrincipal.user_id == session.user.id,
                                                    EventPrincipal.full_access))
```

Reply:
We discussed on irc to skip group-based access and I couldn't figure out the proper query :D' thanks